modules/core/src/main/java/org/apache/ignite/internal/thread/IgniteThreadFactory.java

```diff
@@ -44,7 +44,7 @@ public class IgniteThreadFactory implements ThreadFactory {
     /**
      * Constructor.
      */
-    private IgniteThreadFactory(String nodeName, String poolName, boolean daemon, IgniteLogger log, ThreadOperation[] allowedOperations) {
+    protected IgniteThreadFactory(String nodeName, String poolName, boolean daemon, IgniteLogger log, ThreadOperation[] allowedOperations) {
```

Comment:
Please make the second constructor protected as well. Why don't you use that constructor by the way? It has a full set of fields, unlike this one